File: leetcode1773.go
package main;
/*
* auther yeling
* 
* 
*/
import "fmt";

func countMatches(items [][]string, ruleKey string, ruleValue string) int {
	sum := 0;
	index := 0;
	switch ruleKey {
	case "type":
		index = 0
	case "color":
		index = 1
	case "name":
		index = 2;
	}
	for _,item := range items {
		if item[index] == ruleValue {
			sum++
		}
	}
	return sum
}

var d = map[string]int{"type": 0, "color": 1, "name": 2}

func countMatches2(items [][]string, ruleKey, ruleValue string) (ans int) {
    index := d[ruleKey]
    for _, item := range items {
        if item[index] == ruleValue {
            ans++
        }
    }
    return
}


func main() {
	fmt.Printf("ret %d\n", 10)
}

File: leetcode1773.js
/*
* auther yeling
* 1773. 统计匹配检索规则的物品数量
* 
*/

/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function (items, ruleKey, ruleValue) {
    let sum = 0;
    let index = 0;
    switch (ruleKey) {
        case 'type':
            index = 0;
            break;
        case 'color':
            index = 1;
            break;
        case 'name':
            index = 2;
            break;
    }
    for (let i = 0; i < items.length; i++) {
        if(items[i][index] == ruleValue) {
            sum++;
        }
    }
    return sum;
};



